Numpy/Astropy PyObject `RuntimeWarning` not ignored when testing a single file with Python 3.10

When running pytest on a single file with Python 3.10, I get the following warning: `RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 96 from PyObject`. I seems to be ignored by `pyproject.toml`, but is not ignored when running a single module (e.g. `pytest amical/tests/test_cli.py`). In Python 3.9 everything seems to work as expected.

For context this warning comes (very likely) from a mistake in astropy's compilation, which is not in our control, hence we (try to) ignore it. When you compare a Python 3.10 and a Python 3.9 env, you're actually using different wheels for astropy, and it's possible that only the 3.10 one is broken in that sense.

Now, I don't know why pytest seems to ignore to read the configuration file

here's what I tried
```shell
cd /tmp
git clone git@github.com:neutrinoceros/AMICAL.git
python3.10 -m venv .env
source .env/bin/activate
pip install ".[dev]"
pytest amical/tests/test_io.py
```

And indeed I reproduce the problem you're reporting

<details> <summary> Pytest log </summary>

========================================== test session starts ===========================================
platform darwin -- Python 3.10.1, pytest-7.0.0, pluggy-1.0.0
rootdir: /private/tmp/AMICAL, configfile: pyproject.toml
plugins: cov-3.0.0
collected 0 items / 1 error

================================================= ERRORS =================================================
________________________________ ERROR collecting amical/tests/test_io.py ________________________________
amical/tests/test_io.py:7: in <module>
    import amical
.env/lib/python3.10/site-packages/amical/__init__.py:6: in <module>
    from .analysis.fitting import fits2obs
.env/lib/python3.10/site-packages/amical/analysis/fitting.py:11: in <module>
    from amical.analysis import models
.env/lib/python3.10/site-packages/amical/analysis/models.py:4: in <module>
    from amical.tools import mas2rad
.env/lib/python3.10/site-packages/amical/tools.py:18: in <module>
    from astropy.nddata import Cutout2D
.env/lib/python3.10/site-packages/astropy/nddata/__init__.py:11: in <module>
    from .nddata import *
.env/lib/python3.10/site-packages/astropy/nddata/nddata.py:13: in <module>
    from astropy.wcs.wcsapi import (BaseLowLevelWCS, BaseHighLevelWCS,
.env/lib/python3.10/site-packages/astropy/wcs/__init__.py:26: in <module>
    from .wcs import *
.env/lib/python3.10/site-packages/astropy/wcs/wcs.py:56: in <module>
    from .wcsapi.fitswcs import FITSWCSAPIMixin, SlicedFITSWCS
.env/lib/python3.10/site-packages/astropy/wcs/wcsapi/fitswcs.py:11: in <module>
    from astropy.coordinates import SpectralCoord, Galactic, ICRS
.env/lib/python3.10/site-packages/astropy/coordinates/__init__.py:16: in <module>
    from .builtin_frames import *
.env/lib/python3.10/site-packages/astropy/coordinates/builtin_frames/__init__.py:29: in <module>
    from .fk5 import FK5
.env/lib/python3.10/site-packages/astropy/coordinates/builtin_frames/fk5.py:11: in <module>
    from .utils import EQUINOX_J2000
.env/lib/python3.10/site-packages/astropy/coordinates/builtin_frames/utils.py:16: in <module>
    from astropy.utils import iers
.env/lib/python3.10/site-packages/astropy/utils/iers/__init__.py:2: in <module>
    from .iers import *
.env/lib/python3.10/site-packages/astropy/utils/iers/iers.py:23: in <module>
    from astropy.table import QTable, MaskedColumn
.env/lib/python3.10/site-packages/astropy/table/__init__.py:4: in <module>
    from .column import Column, MaskedColumn, StringTruncateWarning, ColumnInfo
.env/lib/python3.10/site-packages/astropy/table/column.py:22: in <module>
    from ._column_mixins import _ColumnGetitemShim, _MaskedColumnGetitemShim
astropy/table/_column_mixins.pyx:1: in init astropy.table._column_mixins
    ???
E   RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 96 from PyObject
======================================== short test summary info =========================================
ERROR amical/tests/test_io.py - RuntimeWarning: numpy.ndarray size changed, may indicate binary incompa...
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=========================================== 1 error in 21.34s ============================================

</details>

I can also confirm that pytest runs smoothly with a simple invoke.
I thought this could be a regression in pytest 7.0 which just came out a couple days ago so I also tried

```bash
pip install "pytest < 7"
pytest amical/tests/test_io.py
```
but the issue persisted.
Still, this looks like a pytest bug to me, so the next step is to try and reproduce this situation in a much smaller env.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -20,8 +20,8 @@ filterwarnings = [
     "ignore:The distutils.sysconfig module is deprecated, use sysconfig instead:DeprecationWarning",
     # we run tests against a non-GUI backend on purpose
     "ignore:Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.:UserWarning",
-    # we ignore astropy warning (PyObject)
-    "ignore:numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject:RuntimeWarning"
+    # astropy regularly ships wheels that violate numpy ABI incompatibilities, this is out of our control
+    "ignore:numpy.ndarray size changed, may indicate binary incompatibility. Expected \\d\\d from C header, got \\d\\d from PyObject:RuntimeWarning"
 ]
 
 [tool.coverage.run]

